File: UdemyCoursePy/src/mx/irving/stacksandqueuesandmore/Queue2Stacks.py
class Queue2Stacks(object):

    def __init__(self):

        self.stack1 = []
        self.stack2 = []

    def enqueue(self, element):
        self.stack1.append(element)

    def dequeue(self):
        if not self.stack2:
            while self.stack1:
                self.stack2.append(self.stack1.pop())
        return self.stack2.pop()


queue = Queue2Stacks()
for x in range(0, 5):
    queue.enqueue(x)

print(queue.dequeue())
print(queue.dequeue())
print(queue.dequeue())
print(queue.dequeue())
print(queue.dequeue())

File: UdemyCourseKt/src/main/kotlin/mx/irving/stacksandqueues/Queue2Stacks.kt
package mx.irving.stacksandqueues

class Queue2Stacks(
    private val stack1: MutableList<Any> = mutableListOf(),
    private val stack2: MutableList<Any> = mutableListOf()
) {

    fun queue(element: Any) {
        stack1.add(element)
    }

    fun dequeue(): Any? {
        if (stack1.isEmpty() && stack2.isEmpty()) return null
        if (stack2.isEmpty()) {
            while (stack1.isNotEmpty())
                stack2.add(stack1.removeAt(stack1.size - 1))
        }
        return stack2.removeAt(stack2.size - 1)
    }
}
